.github/workflows/publish.yml

```diff
@@ -0,0 +1,24 @@
+name: Publish
+on:
+  release:
+    types: [published]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          registry-url: https://registry.npmjs.org
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Publish to NPM
+        run: npm publish --provenance --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
Before merging this PR a secret called `NPM_TOKEN` should be added to the repository that contains an [access token](https://docs.npmjs.com/creating-and-viewing-access-tokens#creating-access-tokens) for automation.

@mafintosh could I ask you to add this one?